src/test/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceIT.java

```diff
@@ -354,4 +355,33 @@ public final void testUpdateJob(final Vertx aVertx, final VertxTestContext aCont
         }).onFailure(aContext::failNow);
     }
 
+    /**
+     * Tests updating job in db with bad institution ID.
+     *
+     * @param aVertx A Vert.x instance
+     * @param aContext A test context
+     */
+    @Test
+    public final void testUpdateJobBadID(final Vertx aVertx, final VertxTestContext aContext)
+            throws AddressException, MalformedURLException, NumberParseException {
+        final int jobID = 1;
+        final int badInstID = -1;
+        myScheduleStoreProxy.getJob(jobID).onSuccess(original -> {
+            try {
+                final Job modified = new Job(badInstID, new URL("http://new.url.com"), original.getSets().get(),
+                        original.getScheduleCronExpression(), ZonedDateTime.now());
+                myScheduleStoreProxy.updateJob(jobID, modified).onSuccess(result -> {
```

Comment:
Can you make it so updateJob fails if no rows were updated?

Reply:
Added failed future if no update happened